[CI] CoreWithMultipleProjectsClientYamlTestSuiteIT test {yaml=search.vectors/40_knn_search/Dimensions are dynamically set} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #9199 / windows-2025_checkpart2_platform-support-windows](https://gradle-enterprise.elastic.co/s/4wocqi4adbq5a)
- [elasticsearch-periodic #9129 / openjdk23_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/p5jjn7hdbytmo)

**Reproduction Line:**
```
gradlew ":x-pack:qa:multi-project:core-rest-tests-with-multiple-projects:yamlRestTest" --tests "org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT" -Dtests.method="test {yaml=search.vectors/40_knn_search/Dimensions are dynamically set}" -Dtests.seed=473D05C923ECCC5F -Dtests.locale=hr-BA -Dtests.timezone=America/Bahia -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Byaml%3Dsearch.vectors%2F40_knn_search%2FDimensions%20are%20dynamically%20set%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [search.vectors/40_knn_search:632]: field [test_index.mappings.properties.embedding.dims] is null
```

**Issue Reasons:**
- [main] 2 failures in test test {yaml=search.vectors/40_knn_search/Dimensions are dynamically set} (0.6% fail rate in 339 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/40_knn_search.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/40_knn_search.yml
@@ -621,8 +621,14 @@ setup:
       index:
         index: test_index
         id: "0"
+        refresh: true
         body:
           embedding: [ 0.5, 111.3, -13.0, 14.8, -156.0 ]
+
+  # wait and ensure that the mapping update is replicated
+  - do:
+      cluster.health:
+        wait_for_events: languid
 
   - do:
       indices.get_mapping:

